src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -82,24 +88,80 @@ private static String convertToJson(UserAuth userAuth) {
   
   /**
    * If the user is not in database already, add them.
+   * This method returns the Entity object in the database (or the newly-created
+   * user Entity, if none previously existed).
    */
-  public void addUserToDatabase(String email) {
+  public static Entity addUserToDatabase(String email) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Only add User to database if they are not already present there.
+    Entity userEntity = getUserEntityFromEmail(email);
+    if (userEntity == null) {
+      // User is not in database, create and add user object.
+      User newUser = new User(email);
+      Entity newUserEntity = newUser.buildEntity();
+      datastore.put(newUserEntity);
+
+      // Return the new user object.
+      return newUserEntity;
+    }
+
+    // Return the user already in the database.
+    return userEntity;
+  }
+
+  /**
+   * Creates a new UserService object.
+   */
+  public static UserService getUserService() {
+    return UserServiceFactory.getUserService();
```

Comment:
I'm confused as to why this is a method and not just a regular function call. Is there a specific reason why breaking it up was necessary?

Reply:
Yeah I did this for testing purposes, so I didn't have to use PowerMock to mock the result from the `UserServiceFactory.getUserService()`. However, now I think you're likely right -- I should just use PowerMock, so I'll change this.

Reply:
Whatever you think is best! I don't know the pros and cons of using PowerMock very well

Reply:
Changed!